### PR TITLE
Features/expand skeleton stub project

### PR DESCRIPTION
**Pull request by tomling**
- Creation of a new skeleton stub project generated using the default ASP.NET Core 3.1 Web API Template.
- Implemented Swagger by copying in auto-generated files created using API definition provided.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -348,3 +348,4 @@ MigrationBackup/
 
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
```
Comment (SamirHassoun) [inline .gitignore:350]: is this MigrationBackup needed?

Comment (UKHOGreg) [reply .gitignore:350]: I believe all of this gitignore file, except for the last line in the file, are all autogenerated using the Visual Studio template.
There will be a load of irrelevant stuff in there I am sure.
In my view, I would leave it and remove lines only if necessary.

**Review by benbhall — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (SamirHassoun) [reply .gitignore:350]: okey dokey

File: src/UKHO.SalesCatalogueStub.Api/Startup.cs [hunk 2]
```diff
@@ -1,51 +1,131 @@
...
 
         // This method gets called by the runtime. Use this method to add services to the container.
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="services"></param>
         public void ConfigureServices(IServiceCollection services)
         {
+            services
+                .AddMvc(options =>
```
Comment (benbhall) [inline src/UKHO.SalesCatalogueStub.Api/Startup.cs:46]: Behaviour of `AddMvc` is still as it was before Core 3.0 but is a bit heavyweight (equivalent to `AddControllersWithViews()` and `AddRazorPages()`.

Do we not just want `AddControllers()` (which is already present further down)?

File: src/UKHO.SalesCatalogueStub.Api/Startup.cs [hunk 3]
```diff
@@ -1,51 +1,131 @@
...
             app.UseAuthorization();
 
+            app.UseSwagger();
+            app.UseSwaggerUI(c =>
+            {
+                c.SwaggerEndpoint("/swagger/1/swagger.json", "Sales Catalogue Service API Stub");
+
+            });
+
             app.UseEndpoints(endpoints =>
```
Comment (benbhall) [inline src/UKHO.SalesCatalogueStub.Api/Startup.cs:120]: Don't think we want `app.UseEndpoints()` twice? Combine:
```csharp
                .UseEndpoints(endpoints =>
                {
                    endpoints.MapHealthChecks("/health");
                    endpoints.MapControllers();
                });```

Comment (tomling) [reply src/UKHO.SalesCatalogueStub.Api/Startup.cs:120]: Good spot! Fixed.

File: src/UKHO.SalesCatalogueStub.Api/Startup.cs (hunk 2 above)
Comment (benbhall) [reply src/UKHO.SalesCatalogueStub.Api/Startup.cs:46]: DataServices simply has:
```csharp
services.AddControllers().AddNewtonsoftJson(options =>
            {
                options.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
                options.SerializerSettings.Converters.Add(new StringEnumConverter
                {
                    NamingStrategy = new CamelCaseNamingStrategy()
                });
            });
```

Comment (tomling) [reply src/UKHO.SalesCatalogueStub.Api/Startup.cs]: Resolved

**Review by benbhall — APPROVED** [review]

**Review by SamirHassoun — APPROVED** [review]

**Pull request merged**